haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -456,6 +458,19 @@ refsFromRenamed ctx declAlts (hsGroup, _, _, _) =
             mapMaybe (\(L l n) -> give ctx (nameLocToRef n TypeDecl l)) names
         _ -> []
 
+    refsFromImportDecl :: LImportDecl GhcRn -> [Reference]
+    refsFromImportDecl (L _ idecl) = case idecl of
+      ImportDecl {..} ->
+        case ideclHiding of
+          Nothing -> []
+          Just (_, (L _ imports)) -> mapMaybe refsFromImport imports
```

Comment:
Should only do if the (now-omitted) hiding-bool is False. Otherwise we would ref the hidden names.

Bonus: if hiding is True, emit implicit ref for all the (actually used) names. Likely in other PR.

Reply:
Depends on perspective. A hidden name is not really a ref. I wouldn't expect to see `hiding (foobar)` in the backreferences of `foobar`. If the user is really curious about the hidden name, can click on the module to chase it down? I understand though that having a name there but being unclickable is somewhat odd.

For now, you could introduce a constant Bool which controls the behavior and go with whatever default preferred. Later this can be exposed as a config flag if various users prefer differently.

Reply:
Makes sense. I added a boolean flag `generateRefEdgeForHiddenImports`. Does this look good enough?